exp/api/v1beta1/azuremachinepool_default.go

```diff
@@ -18,13 +18,38 @@ package v1beta1
 
 import (
 	"encoding/base64"
+	"fmt"
 
 	"golang.org/x/crypto/ssh"
 	"k8s.io/apimachinery/pkg/util/uuid"
 	infrav1 "sigs.k8s.io/cluster-api-provider-azure/api/v1beta1"
+	azureutil "sigs.k8s.io/cluster-api-provider-azure/util/azure"
 	utilSSH "sigs.k8s.io/cluster-api-provider-azure/util/ssh"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
 )
 
+// SetDefaults sets the default values for an AzureMachinePool.
+func (amp *AzureMachinePool) SetDefaults(client client.Client) {
+	if err := amp.SetDefaultSSHPublicKey(); err != nil {
+		ctrl.Log.WithName("AzureMachinePoolLogger").Error(err, "SetDefaultSshPublicKey failed")
+	}
+
+	machinePool, err := azureutil.FindParentMachinePool(amp.Name, client)
```

Comment:
same comment about retrying, although I think it might be easier to put the retry logic somewhere reusable https://github.com/kubernetes-sigs/cluster-api-provider-azure/pull/3080/files#r1087264640